### DISCUSSION TITLE
Splitting tool

### DISCUSSION THREAD

**Discussion by cymed** [teksi/wastewater#145] [Ideas]
## General information

- Applicant:  @cymed
- Developer: so far @cymed
- User group referent: 
- Developments referent: 

## Users and administrators needs
In the case of changes in gradient and material as well as lateral connections without a manhole, it may be necessary to subsequently separate a section while retaining the sewer information.
In addition, the existence of covered manholes is sometimes only discovered during a TV inspection and the section must be subsequently separated.
The TEKSI Wastewater plugin is therefore to be expanded to include a function that allows the sewer sections to be split and the wastewater node/structure to be recorded at the same time.
see https://github.com/QGEP/QGEP/issues/527 and https://github.com/teksi/wastewater/pull/142

### Graphical identity / Logo

There are two options:

1. Adding a new plugin in analogy to Wizard 
![301883482-65b2a0e3-f545-4c43-ad74-0cd656f9e16b](https://github.com/teksi/wastewater/assets/102035587/817bc6fb-4816-4394-9195-7002d3fc33b7)
![301883728-328f8f77-2a03-46d9-b336-997861b339f9](https://github.com/teksi/wastewater/assets/102035587/7d132a10-0fcb-4b6b-8d23-66d95e777d73)

2. integrate in Wizard by adding a "wastewater node" option and replacing the "wastewater structure" option

### Description of users and administrators feature requirements

After calling the tool, one can add a wastewater node/wastewater structure as per usual. Snapping is enabled for reach vertex and segment. On creation, the elevation is interpolated from the reach or taken from the vertex. After filling out the dialog, the bottom level is extracted and set as reach point level. Subsequently, two new reaches are created.
- The other reach points are kept
- For the wastewater node, all channel and wastewater attributes are kept
- for wastewater structures, the same attributes as with the reach wizard

#### Specific requirements

Necessary features:
### Illustrations of the features and the requirements

_Schemes, screen captures, or any other illustrations._

### Feasibility study

Feasibility, options, alternatives and technical specificities

### Mutualisation 
_Mutualisation and possible impacts with other projects (QGIS, QWAT, QGEP)_

## Specification and offers ( Developer )
Not shared on Github.

### Tenders

Not shared on Github.

### Timeline / deadlines
_Add a detailed timeline_

- [x] 05.02.2024 : First draft
- [ ] 01.10.2021 : 
- [ ] 01.11.2021 : 

- [ ] Warranty period : to be adressed

### Related softwares versions

- QGIS version: 
- Database model version: 
- INTERLIS Model version: 

Public issues links : 
- …

Helpdesk issues links : 
- …

### Realisation dates

- Tender reception date : 
- Tender validation date : 

## Documentation

Documentation links
- User manual : https://teksi.github.io/wastewater/latest/en/user-guide/editing/editing.html#split-a-reach-channel-into-different-reaches and https://teksi.github.io/wastewater/latest/en/user-guide/How-To/index.html#ignoring-reach-points-when-splitting-reaches
- Administrator manual :

## Development and integration

Link and comments from the developer about the product

- Deliverable : 
- Recommendations and comments : 
- Future prospects : 

## Conclusion

Conclusions from applicant, users or admins group

- Conclusion: 
- Future prospects:

**Comment by sjib** [comment]
First developments see https://github.com/teksi/wastewater/pull/142

**Reply by sjib** [reply]
To do: Integrate requirements discussion from https://github.com/teksi/wastewater/pull/142#issuecomment-1936182209 and further

Distinguish different use cases for splitting: PAA / SAA, with or without complete Topology etc.

**Reply by sjib** [reply]
See also VSA Wiki: 
Splitting PAA Channel into two reaches and one channel when PAA channel is beeing added: [de](https://vsa.ch/wiki/erfassung-von-primaeren-knoten-und-haltungen-im-pickellochmodell/) / [fr](https://vsa.ch/fr/wiki/3-2-1-saisie-des-noeuds-et-conduites-primaires-selon-le-modele-du-milieu-du-couvercle/)

<img width="473" height="343" alt="DSS_PAA_Pickellochmodell-3" src="https://github.com/user-attachments/assets/cde69f8b-7fc9-4576-bc5c-bac0ff71b6e8" />
